x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/template/TemplateUtils.java

```diff
@@ -71,34 +79,43 @@ public static String loadTemplate(String resource, String version, String versio
     /**
      * Loads a resource from the classpath and returns it as a {@link BytesReference}
      */
-    public static BytesReference load(String name) throws IOException {
-        return Streams.readFully(TemplateUtils.class.getResourceAsStream(name));
+    public static String load(String name) throws IOException {
+        return Streams.readFully(TemplateUtils.class.getResourceAsStream(name)).utf8ToString();
     }
 
     /**
      * Parses and validates that the source is not empty.
      */
-    public static void validate(BytesReference source) {
+    public static void validate(String source) {
         if (source == null) {
             throw new ElasticsearchParseException("Template must not be null");
         }
+        if (Strings.isEmpty(source)) {
+            throw new ElasticsearchParseException("Template must not be empty");
```

Comment:
nit: I wonder about that: this exception type is used if a REST request is malformed. In this context this makes no sense and we consider the input immutable, right? IMHO this is more of a RTE kind of thing.

Anyway, this is out of scope for this PR, you just add another exception and follow the existing style.